### ISSUE TITLE
Setting to ignore require (only parse import)

### ISSUE THREAD

**Issue by ghetolay** [webpack/webpack#6938]
This is a feature request about a setting for webpack to not handle require statements. I've seen multiples issues about ignoring just one require and the `__non_webpack_require__` solution (which I'm using for the moment) but I'm talking here about **all** `require()`.

Nowadays I've embraced es modules and I'm using `import` for module dependencies, I only use `require` for runtime stuff like loading a config file, a json...
We could also be using `require` to do conditional/dynamic import, and I can see 2 output for that :
1. we want to bundle everything: we could directly use static imports then.
2. we want to keep it dynamic: we'll mark the required module as external (equivalent to not process the require). 

The only problem I can see is usage of third party lib you don't want to be external that still uses require statements. Have to admit I don't have a solution for that at the moment, hoping you'll figure something out :).

So I still think it would be worth having a mode for that, maybe a new `node-es` target.

Is there any other reason to keep using require for compile time that I couldn't think of ?

*For the moment I'm doing a dumb replacement of all `require(` with `__non_webpack_require__(` on a loader (and try not to use the word 'require' in my code) because I want my code to be 'bundler agnostic'.*

**Comment by ooflorent** [comment]
You could disable the parsing of CJS features by setting the type of your JS files to `javascript/esm`:

```js
{
  module: {
    rules: [
      {
        test: /\.js$/,
        type: "javascript/esm",
      }
// …
```

Alternatively, if you use `.mjs` files the type is already set to `javascript/esm`.
Does it fit to your needs?

**Comment by sokra** [comment]
It looks like you just deleted/ignored our lovely crafted [issue template](https://raw.githubusercontent.com/webpack/webpack/master/.github/ISSUE_TEMPLATE.md). It was there for good reasons. Please help us solving your issue by answering the questions asked in this template. I'm closing this. Please open a new issue with filled issue template. Also make sure your issue is not a question. Questions should be posted on Stack Overflow.

**Comment by ghetolay** [comment]
@ooflorent Thks but I got a lot of errors about my imports with that. Good news is that this means this feature is already handled, just a matter of configuration.

I think my request had a bad timing (after v4 release but before docs). I'll wait for the docs to understand what the `type: 'javascript/esm'` fully means/involves. Then I'll probably be able to use it properly to get the desired result.

@sokra I've added your beloved questions before each paragraph.
